File: math/prime/2960.c
#include <stdio.h>

int solution(int N, int K)
{
    int p[1001] = {0, };
    int result = 0;
    int i, j;

    for (i = 2; i <= N; i++) {
        if (p[i] == 0) {
            for (j = i; j <= N; j += i) {
                if (p[j] == 0) {
                    p[j] = 1;
                    result++;

                    if (result == K) {
                        return j;
                    }
                }
            }
        }
    }

    return 0;
}

int main(void)
{
    int N, K;

    scanf("%d %d", &N, &K);
    printf("%d\n", solution(N, K));

    return 0;
}

File: math/prime/2960.py
''' [에라토스테네스의 체] 
'''

import sys

def solution(N: int, K: int) -> int:
    prime = [True] * (N + 1)
    prime[0] = prime[1] = False

    result = 0
    for i in range(2, N + 1):
        if prime[i] != False:
            for j in range(i, N + 1, i):
                if prime[j] == True:
                    prime[j] = False
                    result += 1

                    if result == K:
                        return j
                    
    return -1
                    
N, K = map(int, sys.stdin.readline().split())
sys.stdout.write(str(solution(N, K)) + '\n')